### ISSUE TITLE
Build outputs cannot be shared between CI jobs

### ISSUE THREAD

**Issue by nicholaschiang** [vercel/next.js#22706]
### What version of Next.js are you using?

10.0.8-canary.6

### What version of Node.js are you using?

12.18.3

### What browser are you using?

Firefox 85.0.1 (64-bit)

### What operating system are you using?

Ubuntu 18.04

### How are you deploying your application?

Vercel

### Describe the Bug

I have GitHub Actions setup to:
1. Build the app (in a single job).
2. Run tests in parallel across many jobs (using the build output from the previous job).

To do this, I'm uploading and downloading the `.next` directory between jobs. This is accomplished using [`actions/upload-artifact`](https://github.com/actions/upload-artifact) and [`actions/download-artifact`](https://github.com/actions/download-artifact) like so:
```yaml
build:
  steps:
    - uses: actions/upload-artifact@v2
      with:
        name: build
        path: .next
test:
  strategy:
    matrix:
      browser: [electron, firefox, chrome]
  steps:
    - uses: actions/download-artifact@v2
      with:
        name: build
```

However, this fails because of a readonly file in `.next/cache/next-babel-loader`:
```
Starting download for build
Directory structure has been setup for the artifact
Total number of files that will be downloaded: 1505
events.js:187
      throw er; // Unhandled 'error' event
      ^

Error: EACCES: permission denied, open '/home/runner/work/tutorbook/tutorbook/.next/cache/next-babel-loader/content-v2/sha512/02/61/b70ec8f0caa21b4762eb40df82a733e82935e0e117a47905666fef0368ad34fd0b238b0b52df20bd25e069349ea9a3e0d313560092871c2d9284d053673c'
Emitted 'error' event on WriteStream instance at:
    at internal/fs/streams.js:294:12
    at FSReqCallback.oncomplete (fs.js:146:23) {
  errno: -13,
  code: 'EACCES',
  syscall: 'open',
  path: '/home/runner/work/tutorbook/tutorbook/.next/cache/next-babel-loader/content-v2/sha512/02/61/b70ec8f0caa21b4762eb40df82a733e82935e0e117a47905666fef0368ad34fd0b238b0b52df20bd25e069349ea9a3e0d313560092871c2d9284d053673c'
}
```

### Expected Behavior

You should be able to share build outputs (i.e the `.next` directory) between jobs in a continuous integration environment like GitHub Actions.

And while I don't expect you to solve my particular issue, there should be *some* general documentation on how to share Next.js build outputs between jobs (in continuous integration environments) as I'm sure this is a relatively common use case.

### To Reproduce

Create and run a GitHub Actions workflow with [this (or a similar) configuration](https://github.com/tutorbookapp/tutorbook/actions/runs/615640853/workflow).

**Comment by nicholaschiang** [comment]
I was able to solve this by downloading the artifacts **before** installing dependencies and restoring the cache. Does installing Next.js create a readonly `.next/cache` file for some reason? That would explain this error...

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
